System/InputScheme.cpp

```diff
@@ -236,90 +195,256 @@ namespace RTE {
-				m_InputMappings.at(InputElements::INPUT_L_UP).SetKey(KEY_UP);
-				m_InputMappings.at(InputElements::INPUT_L_DOWN).SetKey(KEY_DOWN);
-				m_InputMappings.at(InputElements::INPUT_L_LEFT).SetKey(KEY_LEFT);
-				m_InputMappings.at(InputElements::INPUT_L_RIGHT).SetKey(KEY_RIGHT);
-				m_InputMappings.at(InputElements::INPUT_R_UP).SetKey(KEY_UP);
-				m_InputMappings.at(InputElements::INPUT_R_DOWN).SetKey(KEY_DOWN);
-				m_InputMappings.at(InputElements::INPUT_R_LEFT).SetKey(KEY_LEFT);
-				m_InputMappings.at(InputElements::INPUT_R_RIGHT).SetKey(KEY_RIGHT);
-				m_InputMappings.at(InputElements::INPUT_FIRE).SetKey(KEY_1_PAD);
-				m_InputMappings.at(InputElements::INPUT_AIM).SetKey(KEY_2_PAD);
-				m_InputMappings.at(InputElements::INPUT_AIM_UP).SetKey(KEY_UP);
-				m_InputMappings.at(InputElements::INPUT_AIM_DOWN).SetKey(KEY_DOWN);
-				m_InputMappings.at(InputElements::INPUT_PIEMENU).SetKey(KEY_3_PAD);
-				m_InputMappings.at(InputElements::INPUT_JUMP).SetKey(KEY_DEL_PAD);
-				m_InputMappings.at(InputElements::INPUT_CROUCH).SetKey(KEY_STOP);
-				m_InputMappings.at(InputElements::INPUT_NEXT).SetKey(KEY_5_PAD);
-				m_InputMappings.at(InputElements::INPUT_PREV).SetKey(KEY_4_PAD);
+			case InputPreset::PresetMouseWASDKeys:
+				m_InputMappings.at(InputElements::INPUT_L_UP).SetKey(KEY_W);
+				m_InputMappings.at(InputElements::INPUT_L_DOWN).SetKey(KEY_S);
+				m_InputMappings.at(InputElements::INPUT_L_LEFT).SetKey(KEY_A);
+				m_InputMappings.at(InputElements::INPUT_L_RIGHT).SetKey(KEY_D);
+				m_InputMappings.at(InputElements::INPUT_FIRE).SetMouseButton(MouseButtons::MOUSE_LEFT);
+				m_InputMappings.at(InputElements::INPUT_AIM).SetPresetDescription("Mouse Move");
+				m_InputMappings.at(InputElements::INPUT_AIM_UP).SetPresetDescription("Mouse Move");
+				m_InputMappings.at(InputElements::INPUT_AIM_DOWN).SetPresetDescription("Mouse Move");
+				m_InputMappings.at(InputElements::INPUT_AIM_LEFT).SetPresetDescription("Mouse Move");
+				m_InputMappings.at(InputElements::INPUT_AIM_RIGHT).SetPresetDescription("Mouse Move");
+				m_InputMappings.at(InputElements::INPUT_PIEMENU).SetMouseButton(MouseButtons::MOUSE_RIGHT);
+				m_InputMappings.at(InputElements::INPUT_JUMP).SetKey(KEY_W);
+				m_InputMappings.at(InputElements::INPUT_CROUCH).SetKey(KEY_S);
+				m_InputMappings.at(InputElements::INPUT_NEXT).SetKey(KEY_E);
+				m_InputMappings.at(InputElements::INPUT_PREV).SetKey(KEY_Q);
+				m_InputMappings.at(InputElements::INPUT_WEAPON_RELOAD).SetKey(KEY_R);
+				m_InputMappings.at(InputElements::INPUT_WEAPON_PICKUP).SetKey(KEY_F);
+				m_InputMappings.at(InputElements::INPUT_WEAPON_DROP).SetKey(KEY_G);
+				m_InputMappings.at(InputElements::INPUT_WEAPON_CHANGE_PREV).SetKey(KEY_1);
+				m_InputMappings.at(InputElements::INPUT_WEAPON_CHANGE_NEXT).SetKey(KEY_2);
+				break;
+			case InputPreset::PresetGenericDPad:
+				// TODO: Don't have any SNES style controllers to test with so no idea what would work or make sense here.
```

Comment:
OpenIssue:tm:.

Reply:
Would be good to have this working as part of this work, so I'd like to deal with it now rather than put in an issue and have it busted for a year.
If you don't wanna make the post and stuff, I can do it if you can tell me what you need.

Reply:
You can make the post and/or the issue. I don't want this insignificant thing holding this back.

Reply:
Like I said, I don't mind kicking it off, but I need to know specifics on what types of gamepads you want a hand with though. And it doesn't really have to hold up this PR, it can be in a quick followup one, it just seems like a shame that you did all this good work to support things well, but half of the gamepads have no decent presets (if nothing else, getting this working will save us people complaining about it all the time like they do with resolutions). Why I say no issue is, again, cause I'd like this to be hopefully sorted out before pre4, and issues have a tendency to sit around for a while.

Also, on a separate note, I'm gonna try to call them gamepads in future to avoid naming collisions with Controller, that I always have trouble with.

Reply:
I guess I need maps for SNES/DS4/XB360 clones and maps for OG SNES/DS4 cause I don't know if clones may have different button/stick/direction layouts.
OG SNES is probably impossibru so I need some reputable brand 12 button gamepad layout. Anything that has more may not work because bumpers would probably be numbered after the extra face buttons.

If they all follow the same standard layout and there's no chance some tree fiddy china DS4 has BUTTON_1 mapped as BUTTON_9 then I may not even need generic presets.
And generally I think the DS4 map is pretty much exactly the same as the XB360 one, but again, I need to be sure they actually use the same button/stick/direction layouts.

Basically I need people to come up with maps that make sense, then map them on a bunch of different clones and compare the settings.ini values with the original.
If they are the same between clones and originals, then that's probably all I need.